piet/src/render_context.rs

```diff
@@ -87,13 +89,15 @@ pub trait RenderContext {
     /// responsiblity? We could have a cache that is flushed when the Direct2D
     /// render target is rebuilt. Solid brushes are super lightweight, but
     /// other potentially retained objects will be heavier.
-    fn solid_brush(&mut self, rgba: u32) -> Result<Self::Brush, Error>;
+    fn solid_brush(&mut self, color: Color) -> Self::Brush;
```

Comment:
What if we removed this API, and just had `Impl Into<Brush> for Color`? I could imagine needing to expose something for gradients (although possibly also not) but I'm not sure that this API is pulling its weight. Is brush creation expensive? Even in that case, we're rarely reusing brushes between draw calls or anything...

Reply:
I have a branch where I'm trying to do this, but it turns out to be not anywhere nearly so easy; the [`SolidColorBrush::create`](https://docs.rs/direct2d/0.2.0/direct2d/brush/struct.SolidColorBrush.html#method.create) method requires a RenderTarget (and there's some funny stuff about the brush being bound to this render target, but I think I can ease that by using a [`DeviceContext`](https://docs.rs/direct2d/0.2.0/direct2d/device_context/struct.DeviceContext.html) everywhere - I tested and this does work on Windows 7 SP1).

I did try making an IntoBrush associated trait that would take either an `&Brush` or a `Color`, but the type tetris got a bit much for me (the result should be a Cow to handle both cases, and needs to encode the lifetime).

Brush creation in D2D is a COM call that does an allocation, so it's not free. It does feel clunky though; a completely reasonable alternative is for the Rust type to be cheap and then have a cache to D2D brush objects. This goes a bit beyond my original intent of being a thin wrapper over D2D, but maybe in this case the simplification for users is worth it.

Reply:
Okay, sounds like future work.